scipy/stats/_continuous_distns.py

```diff
@@ -5438,19 +5438,31 @@ def f3(q, h, k):
                            [q, h, k],
                            default=np.nan)
 
-    def _stats(self, h, k):
-        if h >= 0 and k >= 0:
-            maxr = 5
-        elif h < 0 and k >= 0:
-            maxr = int(-1.0/h*k)
-        elif k < 0:
-            maxr = int(-1.0/k)
-        else:
-            maxr = 5
+    def _get_stats_info(self, h, k):
+        condlist = [
+            np.logical_and(h < 0, k >= 0),
+            k < 0,
+        ]
+
+        def f0(h, k):
+            return (-1.0/h*k).astype(int)
+
+        def f1(h, k):
+            return (-1.0/k).astype(int)
 
-        outputs = [None if r < maxr else np.nan for r in range(1, 5)]
+        return _lazyselect(condlist, [f0, f1], [h, k], default=5)
+
+    def _stats(self, h, k):
+        maxr = self._get_stats_info(h, k)
+        outputs = [None if np.any(r < maxr) else np.nan for r in range(1, 5)]
         return outputs[:]
 
+    def _mom1_sc(self, m, *args):
+        maxr = self._get_stats_info(args[0], args[1])
+        if np.any(m >= maxr):
```

Comment:
```suggestion
        if m >= maxr:
```
If I understand correctly, `m` will always be a scalar when this is called.
If not, then this would not be the correct logic. We would want `np.nan` wherever `m >= maxr` and we would want the integral to be calculated wherever `m < maxr`.

Reply:
> If I understand correctly, `m` will always be a scalar when this is called

It is scalar always